Detectors/CTP/workflowLumi/src/RawToLumiConverterSpec.cxx

```diff
@@ -0,0 +1,183 @@
+  }
+  //
+  std::vector<lumiInfo> lumiPointsHBF1;
+  size_t countsMB = 0;
+  uint32_t payloadCTP;
+  uint32_t orbit0 = 0;
+  bool first = true;
+  gbtword80_t remnant = 0;
+  uint32_t size_gbt = 0;
+  for (auto it = parser.begin(); it != parser.end(); ++it) {
+    auto rdh = it.get_if<o2::header::RAWDataHeader>();
+    auto triggerOrbit = o2::raw::RDHUtils::getTriggerOrbit(rdh);
+
+    auto feeID = o2::raw::RDHUtils::getFEEID(rdh); // 0 = IR, 1 = TCR
+    auto linkCRU = (feeID & 0xf00) >> 8;
+    if (linkCRU == o2::ctp::GBTLinkIDIntRec) {
+      payloadCTP = o2::ctp::NIntRecPayload;
+    } else {
+      continue;
+    }
+    if (first) {
+      orbit0 = triggerOrbit;
+      first = false;
+    }
+    LOG(debug) << "RDH FEEid: " << feeID << " CTP CRU link:" << linkCRU << " Orbit:" << triggerOrbit;
+    // TF in 128 bits words
+    gsl::span<const uint8_t> payload(it.data(), it.size());
+    gbtword80_t gbtWord = 0;
+    int wordCount = 0;
+    std::vector<gbtword80_t> diglets;
+    if (orbit0 != triggerOrbit) {
+      // create lumi per HB
+      lumiInfo lp;
+      lp.mCounts = countsMB;
+      lp.ir.orbit = triggerOrbit;
+      lumiPointsHBF1.push_back(lp);
+      // LOG(info) << "Orbit:" << triggerOrbit << " tvx count:" << countsMB;
+      countsMB = 0;
+      //
+      remnant = 0;
+      size_gbt = 0;
+      orbit0 = triggerOrbit;
+    }
+    for (auto payloadWord : payload) {
+      // LOG(info) << wordCount << " payload:" <<  int(payloadWord);
+      if (wordCount == 15) {
+        wordCount = 0;
+      } else if (wordCount > 9) {
+        wordCount++;
+      } else if (wordCount == 9) {
+        for (int i = 0; i < 8; i++) {
+          gbtWord[wordCount * 8 + i] = bool(int(payloadWord) & (1 << i));
+        }
+        wordCount++;
+        diglets.clear();
+        // LOG(info) << " gbtword:" << gbtWord;
+        o2::ctp::reco_workflow::RawToDigitConverterSpec::makeGBTWordInverse(diglets, gbtWord, remnant, size_gbt, payloadCTP);
+        // count tvx
+        for (auto diglet : diglets) {
+          // LOG(info) << " diglet:" << diglet;
+          gbtword80_t pld = (diglet & mTVXMask);
+          if (pld.count() != 0) {
+            countsMB++;
+          }
+        }
+        gbtWord = 0;
+      } else {
+        // std::cout << "wordCount:" << wordCount << std::endl;
+        for (int i = 0; i < 8; i++) {
+          gbtWord[wordCount * 8 + i] = bool(int(payloadWord) & (1 << i));
+        }
+        wordCount++;
+      }
+    }
+  }
+  lumiInfo lp;
+  lp.mCounts = countsMB;
+  lp.ir.orbit = orbit0;
+  lumiPointsHBF1.push_back(lp);
+  // LOG(info) << "lumiPoints size:" << lumiPointsHBF1.size() << " History size:" << mHistory.size();
+  //
+  size_t tfCounts = 0.;
+  for (auto const& lp : lumiPointsHBF1) {
+    tfCounts += lp.mCounts;
+  }
+  mHistory.push_back(tfCounts);
+  mCounts += tfCounts;
+  // std::cout << tfCounts << " " << mCounts << std::endl;
+  if (mHistory.size() <= mNTFToIntegrate) {
```

Comment:
Why `<=` and not `<` ?

Reply:
when mNHBToIntegrate = 1
mHistory.size() = 1  (was pushed in line 140)
1 < 1  is false(line 150)  and mCounts / nHBIntegrated  are always 0.

Reply:
ok, right